runtime/bindings/python/iree/runtime/io.py

```diff
@@ -64,3 +67,137 @@ def save_archive_file(entries: dict[str, Union[Any, SplatValue]], file_path: Pat
+    if len(array.shape) == 0:
+        flat_array = array.copy()
+    else:
+        flat_array = np.ascontiguousarray(array).view(np.uint8)
+    index.add_buffer(name, flat_array, metadata=metadata)
+
+
+def parameter_index_entry_as_numpy_flat_ndarray(
+    index_entry: ParameterIndexEntry,
+) -> np.ndarray:
+    """Accesses the contents as a uint8 flat tensor.
+
+    If it is a splat, then the tensor will be a view of the splat pattern.
+
+    Raises a ValueError on unsupported entries.
+    """
+    if index_entry.is_file:
+        wrapper = np.array(index_entry.file_view, copy=False)
+    elif index_entry.is_splat:
+        wrapper = np.array(index_entry.splat_pattern, copy=True)
+    else:
+        raise ValueError(f"Unsupported ParameterIndexEntry: {index_entry}")
+
+    return wrapper
+
+
+def parameter_index_entry_as_numpy_ndarray(
+    index_entry: ParameterIndexEntry,
+) -> np.ndarray:
+    """Returns a tensor viewed with appropriate shape/dtype from metadata.
+
+    Raises a ValueError if unsupported.
+    """
+
+    # Decode metadata.
+    metadata = index_entry.metadata.decode()
+    if not metadata.startswith(_metadata_prefix):
+        raise ValueError(
+            f"No metadata for parameter entry {index_entry.key}: Cannot convert to tensor"
+        )
+    metadata = metadata[len(_metadata_prefix) :]
+    d = json.loads(metadata)
+    try:
+        type_name = d["type"]
+        if d["type"] != "Tensor":
+            raise ValueError(
+                f"Metadata for parameter entry {index_entry.key} is not a Tensor ('{type_name}')"
+            )
+        dtype_name = d["dtype"]
+        shape = d["shape"]
+    except KeyError as e:
+        raise ValueError(f"Bad metadata for parameter entry {index_entry.key}") from e
+
+    # Unpack/validate.
+    try:
+        dtype = _NAME_TO_DTYPE[dtype_name]
+    except KeyError:
+        raise ValueError(f"Unknown dtype name '{dtype_name}'")
+    try:
+        shape = [int(d) for d in shape]
+    except ValueError as e:
+        raise ValueError(f"Illegal shape for parameter entry {index_entry.key}") from e
+
+    t = parameter_index_entry_as_numpy_flat_ndarray(index_entry)
+    return t.view(dtype=dtype).reshape(shape)
+
+
+_DTYPE_TO_NAME = (
+    (np.float16, "float16"),
+    (np.float32, "float32"),
+    (np.float64, "float64"),
+    (np.int32, "int32"),
+    (np.int64, "int64"),
+    (np.int16, "int16"),
+    (np.int8, "int8"),
+    (np.uint32, "uint32"),
+    (np.uint64, "uint64"),
+    (np.uint16, "uint16"),
+    (np.uint8, "uint8"),
+    (np.bool_, "bool"),
+    (np.complex64, "complex64"),
+    (np.complex128, "complex128"),
+)
+
+_NAME_TO_DTYPE: dict[str, np.dtype] = {
+    name: np_dtype for np_dtype, name in _DTYPE_TO_NAME
+}
+
+
+def _map_dtype_to_name(dtype) -> str:
+    for match_dtype, dtype_name in _DTYPE_TO_NAME:
+        if match_dtype == dtype:
+            return dtype_name
+
+    raise KeyError(f"Numpy dtype {dtype} not found.")
+
+
+_metadata_prefix = "PYTORCH:"
```

Comment:
Can this be changed to `NUMPY:`? This file doesn't depend on torch.

```suggestion
_metadata_prefix = "NUMPY:"
```

You could also move this up in the file, ahead of its first uses

Reply:
This is essentially our format magic number. Unfortunately, this can not be change if we want files that were already created with IREE Turbine to be compatible.
At some point we could start writing files with a new prefix, but we would have to maintain support for the old one.
I think we would want to have some versioning scheme as well.

Reply:
I changed the format while maintaining support for loading IREE Turbine tensors.